.github/workflows/run-fresh.yml

```diff
@@ -0,0 +1,80 @@
+name: Fresh
+
+on:
+    push:
+        branches:
+            - master
+    pull_request:
+        branches:
+            - "*"
+    schedule:
+        - cron: '0 0 * * *'
+
+jobs:
+    php-lumen-fresh-run:
+        runs-on: ubuntu-latest
+        timeout-minutes: 15
+        env:
+            COMPOSER_NO_INTERACTION: 1
+        strategy:
+            matrix:
+                php: [7.4, 7.3, 7.2]
+                lumen: [7.*, 6.*]
+        name: P${{ matrix.php }} - Lumen${{ matrix.lumen }}
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v2
+              with:
+                  path: src
+            - name: Setup PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: ${{ matrix.php }}
+                  coverage: none
+                  tools: composer:v2
+            - name: Install dependencies
+              run: |
+                  composer create-project --prefer-dist laravel/lumen:${{ matrix.lumen }} --no-progress sample && cd sample
+                  composer require "barryvdh/laravel-ide-helper" --dev --no-update --no-progress
+                  composer require "league/flysystem" --no-update --no-progress
+                  composer update --${{ matrix.dependency-version }} --prefer-dist --no-progress
+            - name: Overwrite source
+              run: mkdir -p sample/vendor/barryvdh/laravel-ide-helper && cp -a src/. sample/vendor/barryvdh/laravel-ide-helper/ && ls sample/vendor/barryvdh/laravel-ide-helper
+            - name: Dump autoloader
+              run: cd sample && composer dump-autoload
+            - name: Insert service provider
+              run: echo '$app->register(\\Barryvdh\\LaravelIdeHelper\\IdeHelperServiceProvider::class);' > /tmp/sp && sed -i "97i $(</tmp/sp)" ./sample/bootstrap/app.php
```

Comment:
I'm not saying this approach is wrong just because https://github.com/rebing/graphql-laravel/blob/19ea3ed120b96c8c9b45398d0fb1ab1514860a4d/tests/integration-lumen.sh#L21 does it different _but_ using sed like this with `97i` I find it's less intuitive in case if the file `bootstrap/app.php` changed.

With the other approach, replacing an existing matching strings, I think it's easier to understand the intention.

Maybe this can be improved even further with the _replacement_ approach, in that it aborts if the string was not replaced, meaning the file changed substantially and it's easier to figure this out?

Same applies to the Laravel below

Reply:
I totally agree